packages/framework/aqueduct/src/request-handlers/requestHandlers.ts

```diff
@@ -65,3 +65,26 @@ export const defaultRouteRequestHandler = (defaultRootId: string) => {
         return undefined; // continue search
     };
 };
+
+/**
+ * Default request handler for a Fluid object that returns the object itself if:
+ *  1. the request url is a "/"
+ *  2. the request url is empty
+ * Returns a 404 error for any other request.
+ */
+export function defaultFluidObjectRequestHandler(fluidObject: IFluidObject, request: IRequest): IResponse {
+    const pathParts = RequestParser.getPathParts(request.url);
+    const requestUrl = (pathParts.length > 0) ? pathParts[0] : request.url;
```

Comment:
I might be missing something, but I believe you can either use pathParts.length === 0 test, or use request.url === "/" || request.url === "". There is no reason to look into pathParts[0], if length is not zero, it's not the droid we are looking for.
Note that testing actual url ensures that it fails if there is any query param, which is likely better default code to have.

Reply:
Is that the right default behavior for PureDataObject's request handler too? I updated that to call into here as well. So, that makes it a breaking change. If a user was expecting a url with query params to return the data object, it will now fail.

Reply:
Hard to say. I like more strict checks in places like that (if you ask for garbage, you get an error). It's like calling a function and supplying different set of arguments form what function expects. in JS, that can go through (extra args are ignored), but in any other environment that's a bug (and quite often - immediate crash :))
So I personally would vote for stronger checks. If data objects wants to allow less strict set of requirements, it can always overwrite this function and do whatever it wants.

Reply:
That's a good point. If you are calling request with query params, you want some special behavior for the query params and should write your own handler. The default handler doesn't know what to do with query params so it returns an error rather than ignoring it and silently returning the object.